internal/engine/upper_test.go

```diff
@@ -61,6 +62,7 @@ func (b *fakeBuildAndDeployer) BuildAndDeploy(ctx context.Context, service model
 }
 
 func (b *fakeBuildAndDeployer) GetContainerForBuild(ctx context.Context, build BuildResult) (k8s.ContainerID, error) {
+	b.getContainerForBuildCount++
 	if build.Image == nil {
```

Comment:
i think the right fix is to make this call b.t.Error

Reply:
ahhhh I understand. Neat!